src/components/ForumDiscussions/ForumPostCard.tsx

```diff
@@ -0,0 +1,325 @@
+// Copyright 2019-2025 @polkassembly/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+import React, { FC, useEffect, useState } from 'react';
+import { Topic } from './types';
+import { Divider } from 'antd';
+import { ClockCircleOutlined, LikeOutlined, EyeFilled } from '@ant-design/icons';
+import { ForumCommentsIcon } from '~src/ui-components/CustomIcons';
+import getRelativeCreatedAt from '~src/util/getRelativeCreatedAt';
+import Link from 'next/link';
+import getCategoryName from './utils/getCategoryName';
+import nextApiClientFetch from '~src/util/nextApiClientFetch';
+import ImageIcon from '~src/ui-components/ImageIcon';
+import formatAvatarUrl from './utils/FormatAvatarUrl';
+import SkeletonAvatar from '~src/basic-components/Skeleton/SkeletonAvatar';
+import SkeletonButton from '~src/basic-components/Skeleton/SkeletonButton';
+import { Pagination } from '~src/ui-components/Pagination';
+import { useTheme } from 'next-themes';
+import { LISTING_LIMIT } from '~src/global/listingLimit';
+
+interface ForumPostCardProps {
+	topics: Topic[];
+}
+
+function formatCount(count: number) {
+	if (count >= 10000) {
+		return (count / 1000).toFixed(1) + 'k';
+	}
+	return count.toString();
+}
+
+const ForumPostCard: FC<ForumPostCardProps> = ({ topics }) => {
+	// const [tagsModal, setTagsModal] = useState<boolean>(false);
+	const { resolvedTheme: theme } = useTheme();
+	const [usernames, setUsernames] = useState<{ [key: number]: string }>({});
+	const [userImg, setUserImg] = useState<{ [key: number]: string }>({});
+	const [isLoading, setIsLoading] = useState<boolean>(false);
+	const [currentPage, setCurrentPage] = useState<number>(1);
+	const itemsPerPage = LISTING_LIMIT;
+
+	const fetchUsernames = async (topicsToFetch: Topic[]) => {
+		setIsLoading(true);
+		const userMap: { [key: number]: string } = {};
+		const userImgMap: { [key: number]: string } = {};
+		for (const topic of topicsToFetch) {
+			try {
+				const { data } = await nextApiClientFetch<any>(`/api/v1/discourse/getTopicById?id=${topic.id}`);
+				const { posts } = data?.data?.post_stream || {};
+				const username = posts?.length > 0 ? posts[0].username || posts[0].name : 'Unknown';
+				const avatarTemplate = posts?.length > 0 ? posts[0].avatar_template : '/path/to/default/avatar.png';
+				userMap[topic.id] = username;
+				userImgMap[topic.id] = formatAvatarUrl(avatarTemplate, '22');
+			} catch (error) {
+				console.error(`Failed to fetch data for topic ID ${topic.id}:`, error);
+				userMap[topic.id] = 'Failed to load';
+				userImgMap[topic.id] = '/path/to/default/avatar.png';
+			}
+		}
+		setUsernames(userMap);
+		setUserImg(userImgMap);
+		setIsLoading(false);
+	};
+
+	useEffect(() => {
```

Comment:
this can call on change of current page
no need for useEffect

Reply:
Yes i know it will reduce the loading time to show the image of username along with his username to 1/3 .

Reply:
No this is a different thing